apps/sqlparser/sqlparser.py

```diff
@@ -0,0 +1,178 @@
+    )
+    fp_out = target_file.relative_to(target_dir())
+
+    if files_to_ignore and str(fp) in files_to_ignore:
+        counters["skipped"] += 1
+        print(f"skipped existing module:{module} name:{fp_out} ")
+        return
+
+    if not dry:
+        # required by shutil.copy2
+        target_file.parent.mkdir(parents=True, exist_ok=True)
+        shutil.copy2(full_fp, target_file)
+
+    result = dict(
+        index=counters["total"],
+        module_index=counters["modules"][str(module)],
+        module=module,
+        database=database,
+        applied=datetime.now().isoformat(timespec="seconds"),
+        fp_in=fp,
+        fp_out=fp_out,
+        target_dir=target_dir(),
+    )
+    counters["written"] += 1
+    counters["total"] += 1
+    counters["modules"][str(module)] += 1
+    print(f"parsed module:{module} name:{fp_out} ")
+    return result
+
+
+def write_report(data):
+    with report_path().open("w") as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=data[0].keys())
+        writer.writeheader()
+        writer.writerows(data)
+
+
+def read_report():
+    # reads report, updates counters, returns report and files_to_ignore
+    global counters
+    if not report_path().exists():
+        return None, None
+
+    with report_path().open() as csvfile:
+        reader = csv.DictReader(csvfile)
+        report = list(reader)
+
+    counters["total"] = len(report)
+
+    files_to_ignore = []
+    for e in report:
+        files_to_ignore.append(e["fp_in"])
+        counters["modules"][e["module"]] += 1
+
+    return report, files_to_ignore
+
+
+def main(dry=False):
+    report, files_to_ignore = read_report()
+    collected = [
+        collect_sql_files(e, files_to_ignore, dry)
+        for e in source_dir().glob(glob)
+    ]
+    collected = [e for e in collected if e is not None]
+    if not dry and collected:
+        write_report(collected)
+
+    print("report stats", counters)
+
+
+def argparsed():
+    global glob, repo_folder, source_suffix, target_suffix, report_suffix
+
+    default = ["--help"]
+    # default = "-y", "", "--glob", "test"
+    parser = argparse.ArgumentParser(
+        description="Collect and copy SQL files to new folder, generate report. Skips collecting pulled updates based on report."
+    )
+    parser.add_argument(
+        "option",
+        nargs="?",
+        choices=["dry-run", "run", None],
+        help="Simulate without changes OR run parser",
+    )
+    parser.add_argument("-y", nargs="?", help="Run without confirmation")
+
+    globals_to_parse = [
+        ["glob", "glob pattern to find scripts"],
+        ["repo_folder", "repository folder"],
+        ["source_suffix", "location of modules folder, appends to repo_folder"],
+        ["target_suffix", "location of target folder, appends to repo_folder"],
+        ["report_suffix", "location of module results, appends to repo_folder"],
+        ["modules_to_ignore", "list of modules to ignore"],
+    ]
```

Comment:
This is better of a list of tuples

```suggestion
    globals_to_parse = [
        ("glob", "glob pattern to find scripts"),
        ("repo_folder", "repository folder"),
        ("source_suffix", "location of modules folder, appends to repo_folder"),
        ("target_suffix", "location of target folder, appends to repo_folder"),
        ("report_suffix", "location of module results, appends to repo_folder"),
        ("modules_to_ignore", "list of modules to ignore"),
    ]
```

better yet though, I'm pretty sure you don't need to do whatever this is doing and you should just `parser.add_argument` for each one of these

Reply:
this is used afterwards to read args too, thats why i decided to decrease boilerplate.

are we still required to drop this?